### DISCUSSION TITLE
Fingerprints and explicit hydrogens

### DISCUSSION THREAD

**Discussion by Jnelen** [rdkit/rdkit#6916] [Q&A] [answered]
Hi all,

I have been playing around with RDKit fingerprints, but I was wondering about the best practices regarding molecule "preparation".
More specifically, is it a good practice to add explicit hydrogens to your molecules (using Chem.AddHs()) before calculating a fingerprint? With morgan fingerprints for instance, I do notice differences in similarity. When I look at tutorials, this doesn't seem to be a common practice. So I was wondering if there is a reason for this, and what the best general practices are?

**Comment by greglandrum** [comment, marked as answer]
Hi @Jnelen, the standard practice is to *not* add the Hs. Most cheminformatics operations are set up to work on hydrogen-suppressed graphs. In most cases the information that the Hs would provide is redundant because the way the heavy atoms are considered in the fingerprint generation includes something about the implicit H count.
